File: classes/parte-5-funciones-y-modularidad/087-visibilidad-encapsulacion-y-contratos-public-private/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// saldo en minúscula: privado del paquete.
type cuenta struct {
	saldo int64
}

func (c *cuenta) depositar(monto int64) {
	c.saldo += monto
}

func (c *cuenta) obtenerSaldo() int64 {
	return c.saldo
}

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.ParseInt(strings.TrimSpace(line), 10, 64)
	c := &cuenta{}
	c.depositar(n)
	c.depositar(n)
	fmt.Printf("saldo=%d\n", c.obtenerSaldo())
}

File: classes/parte-5-funciones-y-modularidad/087-visibilidad-encapsulacion-y-contratos-public-private/implementaciones/python/main.py
import sys


class Cuenta:
    def __init__(self):
        self._saldo = 0  # privado por convención

    def depositar(self, monto):
        self._saldo += monto

    def saldo(self):
        return self._saldo


n = int(sys.stdin.readline())
c = Cuenta()
c.depositar(n)
c.depositar(n)
print(f"saldo={c.saldo()}")
